arches/app/templates/views/components/map-popup.htm

```diff
@@ -9,65 +9,71 @@
+    <!-- ko foreach: popupFeatures -->
+        <!-- ko if: active -->
+            <!-- ko if: displayname -->
+            <div class="hover-feature-title-bar">
+                {% block title %}
+                <div style="display: flex;">
+                    <div data-bind="visible: $parent.popupFeatures.length > 1, click: function(){$parent.advanceFeature('left')}" class="hover-feature-nav-left"><i class="fa fa-angle-left"></i></div>
+                    <div data-bind="visible: $parent.popupFeatures.length > 1, click: function(){$parent.advanceFeature('right')}" class="hover-feature-nav-right"><i class="fa fa-angle-right"></i></div>
+                    <div class="hover-feature-title" data-bind="text: displayname"></div>
+                </div>
+                {% endblock title %}
             </div>
-            <div class="hover-feature-metadata">
-                <span data-bind="text: $root.translations.idString"></span>
-                <span data-bind="text: resourceinstanceid"></span>
+            <!--/ko-->
+            <div class="hover-feature-body">
+                {% block body %}
+                <div class="hover-feature" data-bind="html: map_popup"></div>
+                <!-- ko if: resourceinstanceid -->
+                <div class="hover-feature-metadata-block">
+                    <div class="hover-feature-metadata">
+                        <span data-bind="text: $root.translations.resourceModel"></span>
+                        <span data-bind="text: graph_name"></span>
+                    </div>
+                    <div class="hover-feature-metadata">
+                        <span data-bind="text: $root.translations.idString"></span>
+                        <span data-bind="text: resourceinstanceid"></span>
+                    </div>
+                </div>
+                <!--/ko-->
+                {% endblock body %}
+            </div>
+            <div class="hover-feature-footer">
+                <div>
+                    {% block footer %}
+                    <!-- ko if: resourceinstanceid -->
+                    <a data-bind="click: function () {
+                        window.open(reportURL + ko.unwrap(resourceinstanceid));
+                    }" href="javascript:void(0)">
+                        <i class="ion-document-text"></i>
+                        <span data-bind="text: $root.translations.report"></span>
+                    </a>
+                    <!--/ko-->
+                    <!-- ko if: showEditButton -->
+                    <a data-bind="click: function () {
+                        window.open(editURL + ko.unwrap(resourceinstanceid));
+                    }" href="javascript:void(0)">
+                        <i class="ion-ios-refresh-empty"></i>
+                        <span data-bind="text: $root.translations.edit"></span>
+                    </a>
+                    <!--/ko-->
+                    <!-- ko if: ko.unwrap(geometries).length -->
```

Comment:
I think most of the logic in the map popup will work really well, and allows the logic to be overridden to allow custom geometry retrieval for both resource instances and map overly features. The only catch here is that `geometries` doesn't exist if a geometry from a map overlay is selected so this line throws an exception if it's a map overlay feature and not an Arches resource feature. Can the show/hide logic be implemented in the mapPopupProvider so users that are providing overlay map popups can override this logic?

Also wondering if the visibility of this should also be controlled by an option in the Filter->Select a Filter select box to be consistent with other Map Filter functions.

This maybe be out of scope for this PR, and if so could be followed up by a separate one: Maybe not an issue for geometries that are part of a resource, but filtering by overly-complex geometries in external overlays could potentially cause performance issues (we have a geology layer that would probably be an issue). For overlays layers we could add a configuration variable to the `arches-metadata` map layer object to control whether it's allowed to be used as a source for a map filter geometry.

Reply:
Thanks for this great suggestion, just moved the visibility logic into the map popup provider

Reply:
> Also wondering if the visibility of this should also be controlled by an option in the Filter->Select a Filter select box to be consistent with other Map Filter functions.

That sounds reasonable; my main hesitation is that the UX feels a lot more intuitive to me if I can click a feature and say "I want to search using *this*" vs...
- first going to the map filter
- then selecting a specific filter functionality ("Filter by Popup")
- then selecting the desired popup feature

Reply:
I can buy that argument :).